service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/PertinentInformationObservationValueMapper.java

```diff
@@ -200,4 +215,139 @@ private static boolean isDenominatorPresent(Ratio ratio) {
             && ratio.getDenominator().hasValue()
             && ratio.getDenominator().hasUnit();
     }
+
+    public String mapComponentToPertinentInformation(Observation observation) {
+        String componentPertinentInformation = observation.getComponent()
+            .stream()
+            .map(this::buildComponentPertinentInformation)
+            .filter(StringUtils::isNotEmpty)
+            .collect(Collectors.joining(StringUtils.SPACE));
+
+        return String.format(COMPONENT_TEMPLATE, componentPertinentInformation);
+    }
+
+    private String buildComponentPertinentInformation(Observation.ObservationComponentComponent component) {
+        List<String> componentPertinentInformationList = List.of(
+            extractComponentCode(component),
+            extractComponentValue(component),
+            extractComponentInterpretationCode(component),
+            extractComponentInterpretationText(component),
+            extractComponentReferenceRange(component)
+        );
+
+        var componentText = componentPertinentInformationList.stream()
+            .filter(StringUtils::isNotEmpty)
+            .collect(Collectors.joining(StringUtils.SPACE));
+
+        if (StringUtils.isNotEmpty(componentText)) {
+            return String.format(COMPONENT_DELIMITER_TEMPLATE, componentText);
+        }
+
+        return componentText;
+    }
+
+    private String extractComponentCode(Observation.ObservationComponentComponent component) {
+        if (component.hasCode()) {
+            var code = CodeableConceptMappingUtils.extractTextOrCoding(component.getCode());
+            if (code.isPresent()) {
+                return String.format(COMPONENT_CODE_TEMPLATE, code.get());
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String extractComponentValue(Observation.ObservationComponentComponent component) {
+        if (component.hasValue()) {
+            var value = component.getValue();
+
+            if (COMPONENT_VALUE_FUNCTIONS.containsKey(value.getClass())) {
+                return COMPONENT_VALUE_FUNCTIONS.get(value.getClass()).apply(value);
+            } else {
+                return mapObservationValueToPertinentInformation(value);
+            }
```

Comment:
So this is pretty much `.getOrDefault` but it's a bit awkward because the default doesn't need the key (class)

```suggestion
            return COMPONENT_VALUE_FUNCTIONS.getOrDefault(value.getClass(),
                $ -> this.mapObservationValueToPertinentInformation(value)
            ).apply(value);
```

Your choice